spec/controllers/packages_controller_spec.rb

```diff
@@ -13,66 +13,67 @@
 
   it "should render the index successfully" do
     get :index
-    response.should render_template(:index)
+    expect(response).to render_template(:index)
   end
 
   it "should do a 301 for numerical package ids" do
     get :show, :id => Package.find_by_name("rJython").id.to_s
-    response.should redirect_to("/packages/rJython")
-    response.status.should == "301 Moved Permanently"
+    expect(response).to redirect_to("/packages/rJython")
+    expect(response.status).to eq(301)
   end
 
   it "should do a 301 if package name differs in case from id" do
     get :show, :id => "rjython"
-    response.should redirect_to("/packages/rJython")
-    response.status.should == "301 Moved Permanently"
+    expect(response).to redirect_to("/packages/rJython")
+    expect(response.status).to eq(301)
   end
 
   it "should not incorrectly 301 for package names with dots" do
     get :show, :id => "data-table"
-    response.status.should == "200 OK"
+    expect(response.status).to eq(200)
   end
 
   it "should do a 404 for unknown packages" do
     get :show, :id => "blabla"
-    response.status.should == "404 Not Found"
-    response.should_not be_redirect
+    expect(response.status).to eq(404)
+    expect(response).to_not be_redirect
   end
 
-  it "should redirect if not logged in" do
-    post :toggle_usage, :id => "aaMI"
-    response.should be_redirect
-  end
+  describe 'POST toggle_usage' do
+    context 'with a logged out user' do
+      it "should redirect if not logged in" do
+        post :toggle_usage, :id => "aaMI"
+        expect(response).to be_redirect
+      end
+    end
 
-  it "should be possible to toggle package usage" do
-    login_as_user(:id => 1, :login => "test")
+    context 'with a logged in user' do
+      it "should be possible to toggle package usage" do
+        login_as_user(:id => 1, :login => "test")
 
-    pkg_mock = mock_model(Package)
-    pkg_mock.should_receive(:update_score!)
-    controller.instance_eval do
-      current_user.should_receive(:toggle_usage).with(pkg_mock).and_return(true)
-    end
-    Package.should_receive(:find_by_param).with("aaMI").and_return(pkg_mock)
+        package = FactoryGirl.create(:package, :name => 'aaMI')
+        controller.instance_eval do
+          current_user.should_receive(:toggle_usage).with(package).and_return(true)
```

Comment:
The new syntax is `allow(current_user).to receive(:toggle_user).with(package).and_return(true)`